app/models/custom_filter.rb

```diff
@@ -24,6 +24,31 @@ class CustomFilter < ApplicationRecord
   enum filter_type: { conversation: 0, contact: 1, report: 2 }
   validate :validate_number_of_filters
 
+  def records_count
+    fetch_record_count_from_redis
+  end
+
+  def filter_records
+    user = account.users.try(:first)
```

Comment:
@tejaswinichile I am not sure we can do this.  The conversation permission for various agents will be based on their inbox access, right?  we need to get the Current user here to ensure we are only picking conversations which the agent has access to ?

Reply:
Isn't that for mine-records, should we do that for custom-filter as well?

Reply:
Yep. even in all tabs. agent a might not have access to inbox A., so conversations from inbox A shouldn't be shown on their feed.

Reply:
Note: We have the custom-filter and user association. We only display `user.custom_filters` in the sidebar.